lib/internal/Magento/Framework/Escaper.php

```diff
@@ -309,7 +311,7 @@ public function escapeJs($string)
             $string = (string) $string;
         }
 
-        if ($string === '' || ctype_digit($string)) {
+        if ($string === '' || (is_string($string) && ctype_digit($string))) {
```

Comment:
`is_string` check is redundant because `$string` converted to string above
```
if (!is_string($string)) {
            // In PHP > 8, preg_replace_callback throws an error if the 3rd param type is incorrect.
            // This check emulates an old behavior.
            $string = (string) $string;
        }
```